### DISCUSSION TITLE
hrp startproject fail, 使用python插件脚手架初始化项目报错

### DISCUSSION THREAD

**Discussion by Zzzz-zz** [httprunner/httprunner#1470] [2 - 答疑（Q&A）]
$> hrp startproject demo

10:51PM INF Set log to color console other than JSON format.
10:51PM ??? Set log level
10:51PM INF create new scaffold project force=false pluginType=py projectName=demo
10:51PM INF create folder path=demo
10:51PM INF create folder path=demo/har
10:51PM INF create file path=demo/har/.keep
10:51PM INF create folder path=demo/testcases
10:51PM INF create folder path=demo/reports
10:51PM INF create file path=demo/reports/.keep
10:51PM INF dump data to json path=/Users/zhi.zeng/Documents/Code/CodeTest/hrp/test_01/demo/proj.json
10:51PM INF create file path=demo/.gitignore
10:51PM INF create file path=demo/.env
10:51PM INF create file path=demo/testcases/demo.json
10:51PM INF create file path=demo/testcases/requests.json
10:51PM INF create file path=demo/testcases/requests.yml
10:51PM INF create file path=demo/testcases/ref_testcase.yml
10:51PM INF start to create hashicorp python plugin
10:51PM INF create file path=demo/debugtalk.py
10:51PM INF ensure python3 venv packages=["funppy==v0.5.0","httprunner==v4.2.0"] python3=/Users/zhi.zeng/.hrp/venv/bin/python3
10:51PM INF exec command cmd="/Users/zhi.zeng/.hrp/venv/bin/python3 -m pip --version"
pip 22.0.4 from /Users/zhi.zeng/.hrp/venv/lib/python3.9/site-packages/pip (python 3.9)
10:51PM INF installing python package pkgName=funppy pkgVersion=v0.5.0
10:51PM INF exec command cmd="/Users/zhi.zeng/.hrp/venv/bin/python3 -m pip install --upgrade funppy==v0.5.0 --index-url https://pypi.org/simple --quiet --disable-pip-version-check"
10:51PM ERR create scaffold project failed error="prepare python3 venv failed: pip install funppy==v0.5.0 failed: python package funppy not found"

提示 funppy这个包找不到，但是实际上去到hrp这个虚拟环境下是已经安装了这个包的。

❯ ls -l  /Users/zhi.zeng/.hrp/venv/lib/python3.9/site-packages/| grep funppy
drwxr-xr-x   8 zhi.zeng  staff      256  9  7 22:52 funppy
drwxr-xr-x   8 zhi.zeng  staff      256  9  7 22:52 funppy-0.5.0.dist-info

不知道怎么解决这个问题了，尝试用这个虚拟环境下的python导入 funppy这个包，出现了新的问题
❯ /Users/zhi.zeng/.hrp/venv/bin/python3

Python 3.9.12 (main, Apr  5 2022, 01:52:34)
[Clang 12.0.0 ] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import funppy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/zhi.zeng/.hrp/venv/lib/python3.9/site-packages/funppy/__init__.py", line 3, in <module>
    from funppy.plugin import register, serve
  File "/Users/zhi.zeng/.hrp/venv/lib/python3.9/site-packages/funppy/plugin.py", line 9, in <module>
    import grpc
  File "/Users/zhi.zeng/.hrp/venv/lib/python3.9/site-packages/grpc/__init__.py", line 22, in <module>
    from grpc import _compression
  File "/Users/zhi.zeng/.hrp/venv/lib/python3.9/site-packages/grpc/_compression.py", line 15, in <module>
    from grpc._cython import cygrpc
ImportError: dlopen(/Users/zhi.zeng/.hrp/venv/lib/python3.9/site-packages/grpc/_cython/cygrpc.cpython-39-darwin.so, 0x0002): symbol not found in flat namespace '_CFRelease'

是里面的grpc包调用错了，可是这个包是安装hrp的时候自动下载的。我尝试重新安装hrp，但是还是出现相同的问题。求帮助解决谢谢

**Comment by kvchs** [comment]
执行：
/Users/zhi.zeng/.hrp/venv/bin/python3 -m pip install funppy -i https://pypi.tuna.tsinghua.edu.cn/simple

**Reply by Zzzz-zz** [reply]
It doesn't work. 重新安装清华源的镜像还是有相同的报错问题。

/Users/zhi.zeng/.hrp/venv/bin/python3 -m pip install funppy -i https://pypi.tuna.tsinghua.edu.cn/simple
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Collecting funppy
  Downloading https://pypi.tuna.tsinghua.edu.cn/packages/e2/6b/175de1b4ba541a6b3f3203905a3ec06aad5ab7cff048a3deffc8dc88dd4c/funppy-0.5.0-py3-none-any.whl (9.6 kB)
Requirement already satisfied: grpcio<2.0.0,>=1.44.0 in /Users/zhi.zeng/.hrp/venv/lib/python3.9/site-packages (from funppy) (1.48.1)
Requirement already satisfied: grpcio-tools<2.0.0,>=1.44.0 in /Users/zhi.zeng/.hrp/venv/lib/python3.9/site-packages (from funppy) (1.48.1)
Requirement already satisfied: six>=1.5.2 in /Users/zhi.zeng/.hrp/venv/lib/python3.9/site-packages (from grpcio<2.0.0,>=1.44.0->funppy) (1.16.0)
Requirement already satisfied: protobuf<4.0dev,>=3.12.0 in /Users/zhi.zeng/.hrp/venv/lib/python3.9/site-packages (from grpcio-tools<2.0.0,>=1.44.0->funppy) (3.20.1)
Requirement already satisfied: setuptools in /Users/zhi.zeng/.hrp/venv/lib/python3.9/site-packages (from grpcio-tools<2.0.0,>=1.44.0->funppy) (58.1.0)
Installing collected packages: funppy
Successfully installed funppy-0.5.0
WARNING: You are using pip version 22.0.4; however, version 22.2.2 is available.
You should consider upgrading via the '/Users/zhi.zeng/.hrp/venv/bin/python3 -m pip install --upgrade pip' command.
❯ ls -l | grep funppy
drwxr-xr-x   8 zhi.zeng  staff      256  9 14 23:07 funppy
drwxr-xr-x   8 zhi.zeng  staff      256  9 14 23:07 funppy-0.5.0.dist-info
❯ j demo
/Users/zhi.zeng/Documents/Code/CodeTest/hrp/test_01/demo
❯ cd ..
❯ ls -l
total 0
drwxr-xr-x  9 zhi.zeng  staff  288  9  7 23:08 demo
drwxr-xr-x  8 zhi.zeng  staff  256  9  7 23:15 demo_py
❯ hrp startproject demo_01
11:07PM INF Set log to color console other than JSON format.
11:07PM ??? Set log level
11:07PM INF create new scaffold project force=false pluginType=py projectName=demo_01
11:07PM INF create folder path=demo_01
11:07PM INF create folder path=demo_01/har
11:07PM INF create file path=demo_01/har/.keep
11:07PM INF create folder path=demo_01/testcases
11:07PM INF create folder path=demo_01/reports
11:07PM INF create file path=demo_01/reports/.keep
11:07PM INF dump data to json path=/Users/zhi.zeng/Documents/Code/CodeTest/hrp/test_01/demo_01/proj.json
11:07PM INF create file path=demo_01/.gitignore
11:07PM INF create file path=demo_01/.env
11:07PM INF create file path=demo_01/testcases/demo.json
11:07PM INF create file path=demo_01/testcases/requests.json
11:07PM INF create file path=demo_01/testcases/requests.yml
11:07PM INF create file path=demo_01/testcases/ref_testcase.yml
11:07PM INF start to create hashicorp python plugin
11:07PM INF create file path=demo_01/debugtalk.py
11:07PM INF ensure python3 venv packages=["funppy==v0.5.0","httprunner==v4.2.0"] python3=/Users/zhi.zeng/.hrp/venv/bin/python3
11:07PM INF exec command cmd="/Users/zhi.zeng/.hrp/venv/bin/python3 -m pip --version"
pip 22.0.4 from /Users/zhi.zeng/.hrp/venv/lib/python3.9/site-packages/pip (python 3.9)
11:07PM INF installing python package pkgName=funppy pkgVersion=v0.5.0
11:07PM INF exec command cmd="/Users/zhi.zeng/.hrp/venv/bin/python3 -m pip install --upgrade funppy==v0.5.0 --index-url https://pypi.org/simple --quiet --disable-pip-version-check"
11:07PM ERR create scaffold project failed error="prepare python3 venv failed: pip install funppy==v0.5.0 failed: python package funppy not found"

**Reply by liuxeuman** [reply]
我也是这个问题，这么长时间还没解决办法吗？

**Comment by Minions777** [comment]
遇到同样的问题，现在有解决办法了嘛

**Comment by xifengxixi** [comment]
+10086

**Comment by yxx1411** [comment]
我遇到这样的问题
D:\code\demo>hrp  run ./testcases/requests.yml
12:39PM INF Set log to color console other than JSON format.
12:39PM ??? Set log level
12:39PM INF [init] SetFailfast failfast=true
12:39PM INF [init] SetSaveTests saveTests=false
12:39PM INF [init] SetRequestsLogOn
12:39PM INF load file path="testcases\\requests.yml"
12:39PM INF load file path="D:\\code\\demo\\.env"
12:39PM INF set env key=base_url
12:39PM INF set env key=USERNAME
12:39PM INF set env key=PASSWORD
12:39PM INF load testcases successfully count=1
12:39PM INF start to prepare python plugin output="D:\\code\\demo\\.debugtalk_gen.py" path="D:\\code\\demo\\debugtalk.py"
12:39PM INF exec command cmd="C:\\WINDOWS\\system32\\cmd.exe /c python -m py_compile D:\\code\\demo\\debugtalk.py"
12:39PM INF find all function names functionNames=["get_user_agent","sleep","sum","sum_ints","sum_two_int","sum_two_string","sum_strings","concatenate","setup_hook_example","teardown_hook_example"]
12:39PM INF generate debugtalk success output="D:\\code\\demo\\.debugtalk_gen.py"
12:39PM INF build python plugin successfully output="D:\\code\\demo\\.debugtalk_gen.py" plugin="D:\\code\\demo\\debugtalk.py"
12:39PM INF ensure python3 venv packages=["funppy==v0.5.0"] python3="C:\\Users\\Administrator\\.hrp\\venv\\Scripts\\python.exe"
12:39PM INF exec command cmd="C:\\WINDOWS\\system32\\cmd.exe /c C:\\Users\\Administrator\\.hrp\\venv\\Scripts\\python.exe -m pip --version"
pip 21.1.1 from C:\Users\Administrator\.hrp\venv\lib\site-packages\pip (python 3.8)
12:39PM INF installing python package pkgName=funppy pkgVersion=v0.5.0
12:39PM INF exec command cmd="C:\\WINDOWS\\system32\\cmd.exe /c C:\\Users\\Administrator\\.hrp\\venv\\Scripts\\python.exe -m pip install --upgrade funppy==v0.5.0 --index-url https://pypi.org/simple --quiet --disable-pip
-version-check"
12:40PM ERR python3 venv is not ready error="prepare python3 venv failed: pip install funppy==v0.5.0 failed: python package funppy not found" packages=["funppy==v0.5.0"]
12:40PM ERR [Run] init session runner failed error="init plugin failed: prepare python3 venv failed: pip install funppy==v0.5.0 failed: python package funppy not found"

**Comment by FLuid1900** [comment]
已解決，安裝脚手架時，忽略插件 hrp startproject demo --ignore-plugin
然後按照官方文檔一個一個插件單獨安裝：

Python版本 : Python3.7 / Python3.8 / Python3.9 / Python3.10
依赖库：
| 依赖库 | 版本要求 | 是否必须 | 
| python | 3.7及其以上 | 是 | 
| requests | 2.22.0及其以上 | 是 | 
| pyyaml | 5.4.1及其以上 | 是 | 
| pydantic | 大于1.8.0，小于1.9.0 | 是 |
| loguru | 0.4.1及其以上 | 是 | 
| jmespath | 0.9.5及其以上 | 是 | 
| black | 22.3.0及其以上 | 是 | 
| pytest | 7.1.1及其以上 | 是 |
| pytest-html | 3.1.1及其以上 | 是 | 
| sentry-sdk | 0.14.4及其以上 | 是 |
| allure-pytest | 2.8.16及其以上 | 否 | 
| requests-toolbelt | 0.9.1及其以上   | 否 | 
| filetype | 1.0.7及其以上 | 否 | 
| Brotli | 1.0.9及其以上 | 是 |
| jinja2 | 3.0.3及其以上 | 是 | 
| toml | 0.10.2及其以上 | 是 |
| sqlalchemy | 1.4.36及其以上 | 否 |
